events/abc/client_test.go

```diff
@@ -0,0 +1,717 @@
+						})
+			},
+			checker: func(response *EventNotificationResponse, err error) {
+				assert.Nil(t, response)
+				assert.NotNil(t, err)
+				chkErr := err.(errors.CheckoutAPIError)
+				assert.Equal(t, http.StatusNotFound, chkErr.StatusCode)
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			apiClient := new(mocks.ApiClientMock)
+			credentials := new(mocks.CredentialsMock)
+			environment := new(mocks.EnvironmentMock)
+
+			tc.getAuthorization(&credentials.Mock)
+			tc.apiGet(&apiClient.Mock)
+
+			configuration := configuration.NewConfiguration(credentials, environment, &http.Client{}, nil)
+			eventsClient := NewClient(configuration, apiClient)
+
+			tc.checker(eventsClient.RetrieveEventNotification(tc.eventId, tc.notificationId))
+		})
+	}
+}
+
+func TestRetryWebhook(t *testing.T) {
+	eventId := "evt_az5sblvku4ge3dwpztvyizgcau"
+	webhookId := "wh_az5sblvku4ge3dwpztvyizgcau"
+
+	var (
+		metadataResponse = common.MetadataResponse{
+			HttpMetadata: mocks.HttpMetadataStatusAccepted,
+		}
+	)
+
+	cases := []struct {
+		name             string
+		eventId          string
+		webhookId        string
+		getAuthorization func(*mock.Mock) mock.Call
+		apiPost          func(*mock.Mock) mock.Call
+		checker          func(*common.MetadataResponse, error)
+	}{
+		{
+			name:      "when retry a webhook then return 202 status",
+			eventId:   eventId,
+			webhookId: webhookId,
+			getAuthorization: func(m *mock.Mock) mock.Call {
+				return *m.On("GetAuthorization", mock.Anything).
+					Return(&configuration.SdkAuthorization{}, nil)
+			},
+			apiPost: func(m *mock.Mock) mock.Call {
+				return *m.On("Post", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+					Return(nil).
+					Run(func(args mock.Arguments) {
+						respMapping := args.Get(3).(*common.MetadataResponse)
+						*respMapping = metadataResponse
+					})
+			},
+			checker: func(response *common.MetadataResponse, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, response)
+				assert.Equal(t, http.StatusAccepted, response.HttpMetadata.StatusCode)
+			},
+		},
+		{
+			name: "when request invalid authorization then return error",
+			getAuthorization: func(m *mock.Mock) mock.Call {
+				return *m.On("GetAuthorization", mock.Anything).
+					Return(&configuration.SdkAuthorization{}, nil)
+			},
+			apiPost: func(m *mock.Mock) mock.Call {
+				return *m.On("Post", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+					Return(
+						errors.CheckoutAPIError{
+							StatusCode: http.StatusUnauthorized,
+							Status:     "401 Unauthorized",
+						})
+			},
+			checker: func(response *common.MetadataResponse, err error) {
+				assert.Nil(t, response)
+				assert.NotNil(t, err)
+				chkErr := err.(errors.CheckoutAPIError)
+				assert.Equal(t, http.StatusUnauthorized, chkErr.StatusCode)
+
+			},
+		},
+		{
+			name:      "when event and/or webhook does not exist then return error",
```

Comment:
From the name of this case it looks like it could be splitted in two cases. One for when eventId doesn't exist and another for when the webhookId doesn't exist.

Reply:
👍